//
// Copyright 2020 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

// $title: SGX Identity
// $overview: Protobuf definition for SgxIdentity and related submessages.
// $location: https://asylo.dev/docs/reference/proto/identity/asylo.sgx_identity.v1.html
// $front_matter: redirect_from: /docs/reference/proto/identity/sgx_identity.html

// This package defines messages for SGX identity and identity expectations.
package asylo;

import "asylo/identity/platform/sgx/code_identity.proto";
import "asylo/identity/platform/sgx/machine_configuration.proto";

// A high-level representation of the identity of an SGX enclave, which includes
// the properties of its code identity (`sgx.CodeIdentity`) and the
// security-relevant properties of its execution environment
// (`sgx.MachineConfiguration`).
message SgxIdentity {
  // The enclave's code identity properties. Required.
  optional sgx.CodeIdentity code_identity = 1;

  // Describes the security-relevant configuration of the machine on which the
  // enclave is running. Required.
  optional sgx.MachineConfiguration machine_configuration = 2;
}

// Specification of which fields from `SgxIdentity` to match.
message SgxIdentityMatchSpec {
  // A specification of which fields from `CodeIdentity` to match. Required.
  optional sgx.CodeIdentityMatchSpec code_identity_match_spec = 1;

  // A specification of which fields from `MachineConfiguration` to match.
  // Required.
  optional sgx.MachineConfigurationMatchSpec machine_configuration_match_spec =
      2;  // Required.
}

// A verifier's expectation on an `SgxIdentity`.
message SgxIdentityExpectation {
  // Reference identity matched against the target identity per `match_spec`.
  // Required.
  optional SgxIdentity reference_identity = 1;

  // Specification of which fields from the target enclave identity to match.
  // Required.
  optional SgxIdentityMatchSpec match_spec = 2;
}
